bigtable/client/filters.h

```diff
@@ -0,0 +1,308 @@
+    Filter tmp;
+    tmp.filter_.set_family_name_regex_filter(std::move(pattern));
+    return tmp;
+  }
+
+  /**
+   * Return a filter that accepts only columns matching the given regexp.
+   *
+   * @param pattern the regular expression.  It must be a valid
+   *     [RE2](https://github.com/google/re2/wiki/Syntax) pattern.
+   */
+  static Filter Column(std::string pattern) {
+    Filter tmp;
+    tmp.filter_.set_column_qualifier_regex_filter(std::move(pattern));
+    return tmp;
+  }
+
+  /**
+   * Create a filter that accepts columns in the given range.
+   *
+   * Notice that the range is right-open, i.e., it represents [start,end)
+   */
+  static Filter ColumnRange(std::string begin, std::string end) {
+    Filter tmp;
+    auto& range = *tmp.filter_.mutable_column_range_filter();
+    range.set_start_qualifier_closed(std::move(begin));
+    range.set_end_qualifier_open(std::move(end));
+    return tmp;
+  }
+
+  /**
+   * Return a filter that accepts cells in the given timestamp range.
+   *
+   * The range is right-open, i.e., it represents [start,end)
+   */
+  static Filter TimestampRangeMicros(std::int64_t start, std::int64_t end) {
+    Filter tmp;
+    auto& range = *tmp.filter_.mutable_timestamp_range_filter();
+    range.set_start_timestamp_micros(start);
+    range.set_end_timestamp_micros(end);
+    return tmp;
+  }
+
+  /**
+   * Return a filter that accepts cells in the give timestamp range.
+   *
+   * The range is right-open, i.e., it represents [start,end)
+   *
+   * The function accepts any instantiation of std::chrono::duration<> for the
+   * @p start and @p end parameters.
+   *
+   * @tparam Rep1 the Rep tparam for @p start's type.
+   * @tparam Period1 the Period tparam for @p start's type.
+   * @tparam Rep2 the Rep tparam for @p end's type.
+   * @tparam Period2 the Period tparam for @p end's type.
+   */
+  template <typename Rep1, typename Period1, typename Rep2, typename Period2>
+  static Filter TimestampRange(std::chrono::duration<Rep1, Period1> start,
+                               std::chrono::duration<Rep2, Period2> end) {
+    using namespace std::chrono;
+    return TimestampRangeMicros(duration_cast<microseconds>(start).count(),
+                                duration_cast<microseconds>(end).count());
+  }
+
+  /**
+   * Return a filter that matches keys matching the given regexp.
+   *
+   * @param pattern the regular expression.  It must be a valid RE2 pattern.
+   *     More details at https://github.com/google/re2/wiki/Syntax
+   */
+  static Filter MatchingRowKeys(std::string pattern) {
+    Filter tmp;
+    tmp.filter_.set_row_key_regex_filter(std::move(pattern));
+    return tmp;
+  }
+
+  /**
+   * Return a filter that matches keys matching the given regexp.
+   *
+   * @param pattern the regular expression.  It must be a valid
+   *     [RE2](https://github.com/google/re2/wiki/Syntax) pattern.
+   */
+  static Filter MatchingValue(std::string pattern) { return Filter(); }
```

Comment:
doesn't actually set the filter in the proto

Reply:
Yes, it is coming in the next PR:

https://github.com/coryan/google-cloud-cpp/blob/18b0cb842e8a7c33c5987c58248bf2ae8b7ec31e/bigtable/client/filters.h#L158

Do you mind waiting?